clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -224,7 +224,7 @@
  */
 public class KafkaProducer<K, V> implements Producer<K, V> {
 
-    private static final Logger log = LoggerFactory.getLogger(KafkaProducer.class);
+    private static Logger log;
```

Comment:
You can either make them non-static or pass `Logger` as a parameter. Makes no difference to me, but `log` won't work as a static field when you have multiple instances.